test/test_s2i.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python3
+import logging
+import os
+import time
+
+from conu import DockerBackend, S2IDockerImage, Probe, DockerRunBuilder
+
+import pytest
+
+
+image_name = os.environ.get("IMAGE_NAME", "php")
+test_dir = os.path.abspath(os.path.dirname(__file__))
+test_app_path = os.path.join(test_dir, "test-app")
+app_paths = [
+    test_app_path
+]
+
+
+backend = DockerBackend(logging_level=logging.DEBUG)
+
+
+@pytest.fixture(scope="module", params=app_paths)
+def app(request):
+    i = S2IDockerImage(image_name)
+    app_name = os.path.basename(request.param)
+    app = i.extend(request.param, app_name)
+    yield app
+    pass
+    app.rmi()
+
+
+class GenericTestSuite:
```

Comment:
I'm assuming you want this class to be in conu :)

Reply:
Yes, that was the intention, but I did not go far enough to propose it as a PR.